src/herder/TxSetUtils.h

```diff
@@ -48,5 +50,32 @@ TxSetFrameConstPtr removeTxs(TxSetFrameConstPtr txSet,
 TxSetFrameConstPtr addTxs(TxSetFrameConstPtr txSet,
                           TxSetFrame::Transactions const& newTxs);
 
+// For caching TxSet validity. Consist of {lcl.hash, txSetHash,
+// lowerBoundCloseTimeOffset, upperBoundCloseTimeOffset}
+using TxSetValidityKey = std::tuple<Hash, Hash, uint64_t, uint64_t>;
+
+class TxSetValidityKeyHash
+{
+  public:
+    size_t
+    operator()(TxSetValidityKey const& key) const
+    {
+        SHA256 hasher;
+        hasher.add(std::get<0>(key));
+        hasher.add(std::get<1>(key));
+        xdr::opaque_array<16U> tbs;
+        for (size_t i = 0; i < 8; ++i)
+        {
+            tbs.extend_at(i) = (std::get<2>(key) >> (8 * i)) & 0xff;
+        }
+        for (size_t i = 0; i < 8; ++i)
+        {
+            tbs.extend_at(i + 8) = (std::get<3>(key) >> (8 * i)) & 0xff;
+        }
+        hasher.add(tbs);
+        return std::hash<Hash>()(hasher.finish());
+    }
+};
```

Comment:
It's complicated because you're using crypto hashes.
For hash functions in `unsorted_map`, we use short hashes.

See https://github.com/stellar/stellar-core/blob/9dffed456d837e9e3e0fb937865278562e2671b2/src/ledger/LedgerHashUtils.h#L116 that you modified before.

Reply:
Updated it to use fast hash.

Reply:
Fixed.